relay.go

```diff
@@ -84,6 +89,43 @@ func NewRelay(ctx context.Context, h host.Host, upgrader *tptu.Upgrader, opts ..
 	return r, nil
 }
 
+func (r *Relay) addLiveHop(from, to peer.ID) {
+	r.lhLk.Lock()
+	defer r.lhLk.Unlock()
+
+	r.lhCount++
```

Comment:
Should we be increasing this count only if we weren't tracking this hop already, i.e. (below)

```
if _, ok := tgt[to]; !ok {
  tgt[to] := struct{}{}
  r.lhCount++
}
```

Actually, let me check if the implementation guarantees non-duplicate relays, because if that's the case, we wouldn't get here to begin with.

Reply:
You're right, we don't detect duplicates (which makes *all* this logic incorrect). We should either:

1. Track duplicates in this map (i.e., count connections).
2. Forbid duplicates.

I'd prefer the latter but that may lead to a race condition closing one connection and opening another (the relay may see us trying to keep two connections open. This may not be an issue but (1) will definitely be simpler.